### DISCUSSION TITLE
Hello, i need a random number function

### DISCUSSION THREAD

**Discussion by andyyao2009** [community/community#61923] [Programming Help]
### Select Topic Area

Question

### Body

Hello, i need a random number function in github's README.md document
for example:

```
<p align="center">
	<img alt="img" src="https://www.xxx.com/a.webp?v=1">
</p>

<p align="center">
	<img alt="img" src="https://www.xxx.com/a.webp?v=$RandomNumber">
</p>
```
I want to use this as a counter of git visits.

Or, the user needs these visitor statistics:
IP number
User number
number of visitors
number of pages
weekly
every half month
per month

Thanks.

**Comment by RAY-O-Fn-Sunshine** [comment]
as in phone number?  try text now or whatsapp

**Reply by andyyao2009** [reply]
in github's README.md document

**Comment by BaraaBasim** [comment]
if you want to show the number of visits in your github readme file check out this repository: 
https://github.com/antonkomarev/github-profile-views-counter
should end up with something like this:
```
<img src="https://komarev.com/ghpvc/?username=your-github-username&style=flat-square&color=blue" alt=""/>
```

**Comment by ncerutti** [comment]
Off the top of my head I do not think there is a way of doing that by only editing the README.md file, as it is static.

However, you could use GitHub Actions (https://docs.github.com/en/actions) to periodically change the number that appears in the README.md. If it has to be random, that is pretty straightforward. If it has to be a counter, it gets a bit trickier, but if you can get the visits from the Insights page of your repository, you can insert them in the README.md using GitHub Actions.

**Reply by GhostOf0days** [reply]
You don't need actions. They cause an unnecessary amount of commits to appear. Use a URL where you have dynamically served content and place that URL in your ReadMe.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @andyyao2009 ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by andyyao2009** [comment]
yes ,we need a random number function in README.md document

**Comment by sbarissonmez** [comment]
You can simply use GitHub profile views counter. 

You need to add counter in README.md file in your profile repository via Markdown syntax:

![](https://komarev.com/ghpvc/?username=your-github-username)

> NOTE : Don't forget to replace example your-github-username parameter with real value.


For example for your profile : 

<img align="center" src="https://komarev.com/ghpvc/?username=andyyao2009&style=flat&label=Profile+Views"> 

You can check it this repo > https://github.com/antonkomarev/github-profile-views-counter

**Comment by andyyao2009** [comment]
Thanks for your advice. @sbarissonmez , this is a good way.

This method can get the number of visitors, 

and ,

may be, we better get data analytics.
